projects/graph/graph.py

```diff
@@ -75,27 +75,97 @@ def dft(self, starting_vertex):
                 for edge in self.vertices[node]:
                 ### put them on the stack
                     stack.push(edge)
-    def dft_recursive(self, starting_vertex):
+
+                    
+    def dft_recursive(self, node, visited=set()):
         """
         Print each vertex in depth-first order
         beginning from starting_vertex.
         This should be done using recursion.
         """
-        pass  # TODO
+
+        #take the node (starting_vertex)
+        # if this node hasn't been visited:
+        if node not in visited:
```

Comment:
I will place this condition in the for loop to prevent previously visited nodes from being used recursively.